### ISSUE TITLE
Clear out existing tree on `BlockImportOperation::reset_storage`

### ISSUE THREAD

**Issue by gnunicorn** [paritytech/substrate#1419]
The default implementation of [`BlockImportOperation::reset_storage`](https://github.com/paritytech/substrate/blob/a2dfec4d3c467a481da70efb34f105e141a289fa/core/client/db/src/lib.rs#L332) doesn't clear the existing trie, but it should.

(unearthed through #1418)

**Comment by expenses** [comment]
I'm guessing that this means to clear `self.db_updates` before the rest of the code could potentially return an error? The comment originated in https://github.com/paritytech/substrate/commit/9736c50d88d16abbf1464516e9ea1965812a61c3#diff-ca3049c7e7cc28b90c4b65bf956f7c1cR288 and I'm not sure if it's relevant anymore. @rphmeier

**Comment by rphmeier** [comment]
no, i think what @gnunicorn means is to iterate through the entire trie and kill all of the nodes in the DB.
